### DISCUSSION TITLE
multiplay 에서 player별로 보이는 object를 다르게 할 수 있나요?

### DISCUSSION THREAD

**Discussion by hong99033** [naverz/zepeto-studio-kor#929] [5. Q&A]
안녕하세요. 

multiplay 에서 player 별로 펫을 데리고 다니고 싶은데, 이 펫을 player 본인만 보게 하고 다른 유저들은 못보게 하려고 하는데, 

현재 player를 따라다니는 객체를 프리팹 복사를 통해 생성했는데 이 객체를 다른 player 화면에서 보이는 상황입니다. 

혹시 다른 player들은 보지 못하게, 현재 player 화면에서만 볼 수 있도록 구현이 가능할까요? 가능하다면 어떻게 구현을 해야할까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 본인에게만 보여지는 기능은 클라이언트 단에서만 구현하시면 됩니다.
지금은 서버에 동기화 되도록 구현하셔서 다 보여지는 것 같습니다 :)

**Reply by capgreen02** [reply]
저도 이 부분이 궁금한데 클라이언트 단이 정확히 무엇을 말하는지 알 수 있을까요??

**Reply by Zepeto-TechEvangelist** [reply]
서버와 통신하지 않고 로컬에서만 실행되는 스크립트를 뜻합니다.
예를 들어 동기화가 반드시 필요한 플레이어간의 위치같은 정보는 서버-클라이언트가 통신이 되도록 구현해야 합니다.

반면 동기화가 필요 없는 꾸밈용 오브젝트나 윗 분이 예시로 들어주신 내 눈에만 보이는 펫의 경우는
일반적인 싱글플레이 스크립트처럼 구현하면 됩니다.

**Reply by hong99033** [reply]
죄송합니다. 이해가 잘 안가서요ㅠㅜ

예제에 올려주신 멀티플레이 영상을 보고 멀티플레이 환경을 구현을 했는데 여기서는 스키마에 값 추가를 했었는데요, 싱글플레이 스크립트처럼 구현하라는 말씀은 스키마 값 추가 없이 그냥 유니티에 객체생성을 하면 되는 것을 의미하시는 건가요?

**Reply by Zepeto-TechEvangelist** [reply]
네 맞습니다.

**Reply by Zepeto-TechEvangelist** [reply]
최근 공개된 https://github.com/naverz/zepeto-studio-kor/discussions/902 샘플에서 싱글플레이 씬과 멀티플레이 씬이 각각 제공되고 있으니 한번 보시면서 차이점을 분석하시는 것도 좋을 것 같습니다 :)
